tools/utils/gh.py

```diff
@@ -0,0 +1,360 @@
+        return ExitCode.FAIL
+
+    if found_artifact_path.suffix == ".zip":
+        with zipfile.ZipFile(found_artifact_path) as zfile:
+            zfile.extractall(path=artifacts_path)
+    else:
+        ctx.run("tar", "xf", found_artifact_name, cwd=artifacts_path)
+
+    return ExitCode.OK
+
+
+def download_nox_artifact(
+    ctx: Context,
+    run_id: int = None,
+    slug: str = None,
+    nox_env: str = "ci-test-onedir",
+    repository: str = "saltstack/salt",
+) -> ExitCode:
+    """
+    Download CI nox artifacts.
+    """
+    if TYPE_CHECKING:
+        assert run_id is not None
+        assert slug is not None
+
+    artifacts_path = tools.utils.REPO_ROOT / ".nox" / nox_env
+    if artifacts_path.exists():
+        ctx.error(
+            f"The '.nox/{nox_env}' directory already exists ... Stopped processing."
+        )
+        return ExitCode.SOFT_FAIL
+    artifact_name = f"nox-{slug}-{nox_env}"
+    found_artifact_name = download_artifact(
+        ctx,
+        dest=tools.utils.REPO_ROOT,
+        run_id=run_id,
+        artifact_name=artifact_name,
+        repository=repository,
+    )
+    nox = shutil.which("nox")
+    if nox is None:
+        ctx.error("Could not find the 'nox' binary in $PATH")
+        return ExitCode.FAIL
+    ret = ctx.run(nox, "-e", "decompress-dependencies", "--", slug, check=False)
+    if ret.returncode:
+        ctx.error("Failed to decompress the nox dependencies")
+        return ExitCode.FAIL
+    return ExitCode.OK
+
+
+def download_pkgs_artifact(
+    ctx: Context,
+    run_id: int = None,
+    slug: str = None,
+    arch: str = "x86_64",
+    repository: str = "saltstack/salt",
+) -> ExitCode:
+    """
+    Download CI nox artifacts.
+    """
+    if TYPE_CHECKING:
+        assert run_id is not None
+        assert slug is not None
+
+    artifact_name = "salt-*-"
+    if "windows" in slug:
+        if arch in ("x64", "x86_64"):
+            ctx.info(f"Turning passed arch {arch!r} into 'amd64'")
+            arch = "amd64"
+        if arch not in ("amd64", "x86"):
+            ctx.error(
+                "The allowed values for '--arch' on Windows are 'amd64' and 'x86'"
+            )
+            return ExitCode.FAIL
+        artifact_name += f"{arch}-MSI"
+    else:
+        if arch == "arm64":
+            ctx.info(f"Turning passed arch {arch!r} into 'aarch64'")
+            arch = "aarch64"
+        elif arch == "x64":
+            ctx.info(f"Turning passed arch {arch!r} into 'x86_64'")
+            arch = "x86_64"
+        if arch not in ("x86_64", "aarch64"):
+            ctx.error(
+                f"The allowed values for '--arch' for {slug} are 'x86_64', 'aarch64' or 'arm64'"
+            )
+            return ExitCode.FAIL
+
+        if slug.startswith(("debian", "ubuntu")):
+            artifact_name += f"{arch}-deb"
+        elif slug.startswith(
+            ("almalinux", "amazonlinux", "centos", "fedora", "opensuse", "photonos")
```

Comment:
> No Rocky Linux ?, and wondering about 'RedHat', if there is an actual redhat rather than relying on centos

This has to respect the OS's we test, and the "slugs" we use for them.
We don't test rocky, nor redhat.